src/v/redpanda/services.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+#include "cloud_storage/fwd.h"
+#include "cluster/fwd.h"
+#include "kafka/server/fwd.h"
+#include "raft/fwd.h"
+#include "seastarx.h"
+#include "storage/fwd.h"
+
+#include <seastar/core/sharded.hh>
+
+struct services {
```

Comment:
wouldn't you want to reduce dependencies, etc by creating one `services` per subsystem to define API boundaries clearly. I think i see this as cyclical dependency problems.

Reply:
Last time I ended up using grep/ag and drawing a picture for understanding the dependency graph (there are probably better tools--tips welcomed). The fwd.h files are all leaves in the graph, right, so this creates a bunch of edges into services.h which has edges pointing to these fwd.h declarations. Where is the cycle?

Reply:
I think I understand the point about a bigger dependency graph for all the things that reference services (though the "vertices" are small at the end of the paths here).. just missed the point about leading to cycles

Reply:
The point about API separation is interesting.. I guess I think of this as a application-scope (and lifetime) container.. and application is the thing that composes all these APIs

Reply:
i mean easy to create a cycle. as in, if you consume 1 struct, and say pass it to storage, now storage depends on controller and controller depends on storage

Reply:
In order to create a dependency one have to use the `services`instance and actually include proper headers (e.g. I have to include `cluster/metadata_cache.h` to add `metadata_cache` dependency). As a dependency breaker it should work fine.

But having the `services` object split by subsystems will also be beneficial. It will be possible to see that some service depends on `cluster_services` and `storage_services` and not just `services`. It improves readability and maintainability a bit.